modules/sirdataRtdProvider.js

```diff
@@ -0,0 +1,402 @@
+  moduleConfig.params = moduleConfig.params || {};
+
+  var tcString = (userConsent && userConsent.gdpr && userConsent.gdpr.consentString ? userConsent.gdpr.consentString : '');
+  var gdprApplies = (userConsent && userConsent.gdpr && userConsent.gdpr.gdprApplies ? userConsent.gdpr.gdprApplies : '');
+
+  moduleConfig.params.partnerId = moduleConfig.params.partnerId ? moduleConfig.params.partnerId : 1;
+  moduleConfig.params.key = moduleConfig.params.key ? moduleConfig.params.key : 1;
+
+  var sirdataDomain;
+  var sendWithCredentials;
+
+  if (userConsent.coppa || (userConsent.usp && (userConsent.usp[0] == '1' && (userConsent.usp[1] == 'N' || userConsent.usp[2] == 'Y')))) {
+    // if children or "Do not Sell" management in California, no segments, page categories only whatever TCF signal
+    sirdataDomain = 'cookieless-data.com';
+    sendWithCredentials = false;
+    gdprApplies = null;
+    tcString = '';
+  } else if (getGlobal().getConfig('consentManagement.gdpr')) {
+  // Default endpoint is cookieless if gdpr management is set. Needed because the cookie-based endpoint will fail and return error if user is located in Europe and no consent has been given
+    sirdataDomain = 'cookieless-data.com';
+    sendWithCredentials = false;
+  }
+
+  // default global endpoint is cookie-based if no rules falls into cookieless or consent has been given or GDPR doesn't apply
+  if (!sirdataDomain || !gdprApplies || (utils.deepAccess(userConsent, 'gdpr.vendorData.vendor.consents') && userConsent.gdpr.vendorData.vendor.consents[53] && userConsent.gdpr.vendorData.purpose.consents[1] && userConsent.gdpr.vendorData.purpose.consents[4])) {
+    sirdataDomain = 'sddan.com';
+    sendWithCredentials = true;
+  }
+
+  var actualUrl = moduleConfig.params.actualUrl || getRefererInfo().referer;
+
+  const url = 'https://kvt.' + sirdataDomain + '/api/v1/public/p/' + moduleConfig.params.partnerId + '/d/' + moduleConfig.params.key + '/s?callback=&gdpr=' + gdprApplies + '&gdpr_consent=' + tcString + (actualUrl ? '&url=' + actualUrl : '');
+  ajax(url, {
+    success: function (response, req) {
+      if (req.status === 200) {
+        try {
+          const data = JSON.parse(response);
+          if (data && data.segments) {
+            addSegmentData(adUnits, data, moduleConfig, onDone);
+          } else {
+            onDone();
+          }
+        } catch (e) {
+          onDone();
+          utils.logError('unable to parse Sirdata data' + e);
+        }
+      } else if (req.status === 204) {
+        onDone();
+      }
+    },
+    error: function () {
+      onDone();
+      utils.logError('unable to get Sirdata data');
+    }
+  },
+  null,
+  {
+    contentType: 'text/plain',
+    method: 'GET',
+    withCredentials: sendWithCredentials,
+    referrerPolicy: 'unsafe-url',
+    crossOrigin: true
+  });
+}
+
+export function setGlobalOrtb2(segments, categories) {
+  try {
+    let addOrtb2 = {};
+    let testGlobal = getGlobal().getConfig('ortb2');
```

Comment:
@nouchy A few things about this. When you run getConfig passing in the property as you have here, it returns the value if exists, meaning that testGlobal should never have the ortb2 property. To simplify this, I would suggest the following:

`let testGlobal = getGlobal().getConfig('ortb2') || {};`

This way testGlobal is either the value of ortb2 or an empty object. Then you do not need check each time for its validity. i.e. line 99 could then be simplified to the following:

`let ortb2 = {ortb2: utils.mergeDeep({}, testGlobal, addOrtb2)};`

Reply:
Done :) Thank you !